src/browser/of_events.ts

```diff
@@ -39,9 +41,36 @@ class OFEvents extends EventEmitter {
                 // Wildcard on any channel/topic of a specified source (ex: 'window/*/myUUID-myWindow')
                 super.emit(route(channel, '*', source), envelope);
             }
+            if (channel === 'window') {
+                const propTopic = `window-${topic}`;
+                const dontPropagateToSystem = [
+                    'auth-requested'
+                ];
+                eventPropagations.set(route.application(propTopic, uuid), {...payload, type: propTopic, topic: 'application'});
+                if (!dontPropagateToSystem.some( x => x === topic)) {
+                   eventPropagations.set(route.system(propTopic), {...payload, type: propTopic, topic: 'system'});
+                }
+            } else if (channel === 'application') {
+                const propTopic = `application-${topic}`;
+                const appWindowEventsNotOnWindow = [
+                    'window-alert-requested',
+                    'window-created',
+                    'window-end-load',
+                    'window-responding',
+                    'window-start-load'
+                ];
+                if (!topic.match(/^window-/)) {
+                    eventPropagations.set(route.system(propTopic), { ...payload, type: propTopic, topic: 'system' });
+                } else if (appWindowEventsNotOnWindow.some(t => t === topic)) {
+                    eventPropagations.set(route.system(topic), {...payload, type: propTopic, topic: 'system'});
+                }
+            }
         }
-
-        return super.emit(routeString, ...data);
+        const result = super.emit(routeString, ...data);
+        eventPropagations.forEach((payload, eventString) => {
+            this.emit(eventString, payload, ...extraArgs);
```

Comment:
Isn't this the same as `...data`?

Reply:
No, I'm changing the type and topic on the payload when I set it in the map.

Reply:
Suggest renaming `payload` in the `forEach` lambda to avoid confusion. Perhaps `propagationPayload` or similar...

Reply:
Also, does the event propagation not affect the `result` return value here? Function `this.emit(...)` apparently returns something, but the value is discarded.